src/lib.rs

```diff
@@ -687,6 +703,27 @@ impl<'a> ParserState<'a> {
         }
     }
 
+    /// An MD5 mangled name is `??@` followed by 32 characters and a terminating `@`.
+    ///
+    /// See https://github.com/llvm/llvm-project/blob/818cf30b83305fa4a2f75821349210b0f7aff4a4/llvm/lib/Demangle/MicrosoftDemangle.cpp#L754
+    fn read_md5_name(&mut self) -> Result<Name<'a>> {
+        let start_offset = self.offset;
+
+        if start_offset < 3 || !self.input[start_offset - 3..].starts_with("??@") {
+            return Err(self.fail("expected MD5 mangled name"));
+        }
+
+        while !self.consume(b"@") {
+            self.advance(1);
+        }
```

Comment:
Like you said in https://github.com/getsentry/symbolic/pull/338#discussion_r591172001 , I think it would be a good idea to check that these are hexadecimal characters.